### PR TITLE
Add links between format_args! macro and std::fmt::Arguments struct

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @GuillaumeGomez

### CODE REVIEW

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -272,10 +272,12 @@ impl<'a> Arguments<'a> {
 /// safely be done so, so no constructors are given and the fields are private
 /// to prevent modification.
 ///
-/// The `format_args!` macro will safely create an instance of this structure
+/// The [`format_args!`] macro will safely create an instance of this structure
 /// and pass it to a function or closure, passed as the first argument. The
 /// macro validates the format string at compile-time so usage of the `write`
 /// and `format` functions can be safely performed.
+///
+/// [`format_args!`]: ../../std/macro.format_args!.html
```
Comment (Mark-Simulacrum) [inline src/libcore/fmt/mod.rs]: Should we link to `../../std/macro.format_args!.html` or `../../std/macro.format_args.html`? (note the lack of the `!` in the second version).

The latter seems to not work, but I might be doing something wrong....

Comment (GuillaumeGomez) [inline src/libcore/fmt/mod.rs]: I would say the second.

File: src/libstd/macros.rs [hunk 2]
```diff
@@ -193,12 +193,15 @@ macro_rules! assert_approx_eq {
 pub mod builtin {
     /// The core macro for formatted string creation & output.
     ///
-    /// This macro produces a value of type `fmt::Arguments`. This value can be
-    /// passed to the functions in `std::fmt` for performing useful functions.
+    /// This macro produces a value of type [`fmt::Arguments`]. This value can be
+    /// passed to the functions in [`std::fmt`] for performing useful functions.
     /// All other formatting macros (`format!`, `write!`, `println!`, etc) are
```
Comment (GuillaumeGomez) [inline src/libstd/macros.rs]: Why not adding urls for these macros as well?

File: src/libcore/fmt/mod.rs [hunk 3]
```diff
@@ -272,10 +272,12 @@ impl<'a> Arguments<'a> {
 /// safely be done so, so no constructors are given and the fields are private
 /// to prevent modification.
 ///
-/// The `format_args!` macro will safely create an instance of this structure
+/// The [`format_args!`] macro will safely create an instance of this structure
 /// and pass it to a function or closure, passed as the first argument. The
 /// macro validates the format string at compile-time so usage of the `write`
```
Comment (GuillaumeGomez) [inline src/libcore/fmt/mod.rs]: Why not adding urls for write and format functions?

**Pull request merged**